broadinstitute/pearl
# JN-537 adding api wrapper utils

Adds some hooks and utils for standard loading and error handling for our admin pages.  I've updated a representative sample of existing components with the new utility functions.  If y'all like this direction, I'll apply them to all the other relevant places where we call APIs.

TO TEST:
1. go to https://localhost:3000/ourhealth/studies/ourheart/env/sandbox/mailingList
2. confirm it loads, behaves correctly on environment switch, and updates after deleting an entry.
3. update MailingListController's get() method to throw an error (e.g. by appending a random string to the portal shortcode)
4. reload the mailing list view page.
5. confirm you see an error alert
<img width="995" alt="image" src="https://github.com/broadinstitute/juniper/assets/2800795/a8f31063-a1e5-4cb1-932f-5bf6859c24c5">
